haystack/components/generators/chat/openai.py

```diff
@@ -416,14 +454,35 @@ def _prepare_api_call(  # noqa: PLR0913
 
         is_streaming = streaming_callback is not None
         num_responses = generation_kwargs.pop("n", 1)
+
```

Comment:
Do you think that it would make sense to do something as follows to reduce duplication?

```python
base_args = {
  "model": self.model,
   "messages": openai_formatted_messages,
   "n": num_responses,
   **openai_tools,
   **generation_kwargs}
```